Change the locally defined Db's to use aliases from the ipmiComm db's

gustavosr8 opened the pull request:
Since the defined Db's are just copies of ipmiComm db's, with the name changed to fit the LNLS name convention, we can replace the record definition with aliases to the support module record.

henriquesimoes commented on ipmiMgrApp/Db/cooling_unit.substitutions:
```diff
@@ -23,19 +23,48 @@
+            { $(id) , $(fruid) }
+}
+
+file "$(IPMI_COMM)/Db/sensor_ai.db"
+{
+    pattern { attr         , sensinst    , type    , fruid    , prefix }
+            { LM75Temp1    , 1           , 1       , $(fruid) , $(id)  }
+            { LM75Temp2    , 2           , 1       , $(fruid) , $(id)  }
+            { 3V3          , 1           , 2       , $(fruid) , $(id)  }
+            { 12V          , 2           , 2       , $(fruid) , $(id)  }
+            { 12V          , 3           , 2       , $(fruid) , $(id)  }
+            { Fan1         , 1           , 4       , $(fruid) , $(id)  }
+            { Fan2         , 2           , 4       , $(fruid) , $(id)  }
+            { Fan3         , 3           , 4       , $(fruid) , $(id)  }
+            { Fan4         , 4           , 4       , $(fruid) , $(id)  }
+            { Fan5         , 5           , 4       , $(fruid) , $(id)  }
+            { Fan6         , 6           , 4       , $(fruid) , $(id)  }
+}
+
+file "sensor_ai_alias.db"
```
I really don't know if this pays off, but is it possible to have a substitution file that substitutes the filename from the `file` clause while keeping anything else untouched?

This could avoid having aliases duplicating all expansions. Surely would increase significantly the amount of files and complexity of the substitutions graph.

gustavosr8 replied:
Unfortunately, the substitutions file only generates `.db` files, so it's not possible to expand a `.substitutions` macro in another `.substitutions` file.

henriquesimoes commented on ipmiMgrApp/Db/cooling_unit.substitutions:
```diff
@@ -49,4 +78,5 @@ file "sensor_ai.db"
-}
\ No newline at end of file
+}
+
```
Nit: useless empty.

henriquesimoes commented on ipmiMgrApp/Db/sensor_ai_alias.template:
```diff
@@ -0,0 +1,3 @@
+alias("$(dev):$(prefix)$(attr)$(sensinst)", "$(P)$(R)$(prefix)$(attr)-Mon")
+
+alias("$(dev):$(prefix)$(attr)$(sensinst)P", "$(P)$(R)$(prefix)$(attr)Prs-Sts")
```
Missing `$(sensinst)` on the alias name. As we discussed in person, mention this fixes a bug in the previous implementation where records with the same name would be generated and one would overwrite the other.

ericonr replied:
Honestly, I think most of our IOCs should use [`dbRecordsOnceOnly = 1`](https://github.com/lnls-dig/afc-epics-ioc/blob/master/iocBoot/iocutca/common.cmd#L14).

It only allows records to be defined once, so if your substitutions have any issues, the IOC warns you.

henriquesimoes commented on iocBoot/iocIpmiMgr/stIpmiMgr.cmd:
```diff
@@ -13,9 +13,8 @@ drvAsynIPPortConfigure ("$(PORT)","$(IPADDR):623 udp",0,0,0)
-dbLoadRecords("$(ASYN)/db/asynRecord.db","P=${P}, R=${R}asyn,PORT=$(PORT),ADDR=0,OMAX=80,IMAX=80")
-dbLoadRecords("${TOP}/db/microtca_bpm_crate.db", "link=$(PORT),P=$(P),R=$(R)")
-
+dbLoadRecords("$(TOP)/db/asynRecord.db","P=${P}, R=${R}asyn,PORT=$(PORT),ADDR=0,OMAX=80,IMAX=80")
+dbLoadRecords("${TOP}/db/microtca_bpm_crate.db", "link=$(PORT),P=$(P),R=$(R),dev=$(P)$(R),unit='',aliasprefix=''")
```
Maybe we can have right-sided `$(P)` (or `$(R)`) defined without the final `:` so that the `dev` does not contain duplicated `::`.

ericonr reviewed (commented):
I wonder if, in order to avoid duplicating the entries in `substitutions`, you couldn't have the alias files `include` the upstream DBs before defining their aliases.

henriquesimoes commented on ipmiMgrApp/Db/Makefile:
```diff
@@ -36,6 +34,9 @@ DB += rtm_lamp.db
 
+
+#Modules
+DB_INSTALLS += $(ASYN)/db/asynRecord.db
```
This change does not belong in this PR, IMHO. b51ac623288a7ac500bec726aeb51cb6483dcd4e already does this in the appropriate context.

henriquesimoes commented on ipmiMgrApp/Db/afcv3.substitutions:
```diff
@@ -53,4 +54,5 @@ file "sensor_ai.db"
-}
\ No newline at end of file
+}
+
```
Nit: _remove_ extra empty line. You may want to check this in the other files as well; several of them have extra empty line where shouldn't be needed. Not sure if my previous comment on the topic was ambiguous about adding or removing them. Sorry if so.

henriquesimoes commented on ipmiMgrApp/Db/afcv3.substitutions:
```diff
@@ -27,13 +27,14 @@
 
-file "fru_common.db"
+
+file "fru_basic_alias.db"
```
Doesn't `fru_basic_alias.db` already include the expansion of `sensor_ai.db`, through the expansion of the upstream [`fru_basic.template`][1]?

As far as I can understand, the previous `fru_common.db` would not include them. Am I missing something?

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_basic.substitutions#L32

henriquesimoes commented on ipmiMgrApp/Db/fru_cu_alias.template:
```diff
@@ -0,0 +1,7 @@
+include "fru_cu.db"
```
Isn't it [`fru_cu_common.db`][1] in the upstream?

I think, in the end, this is basically the same question I've made regarding `fru_basic_alias`, since `fru_cu.db` includes the substituted records from `sensor_ai.db` while `fru_cu_common.db` does not.

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_cu_common.db

henriquesimoes commented on ipmiMgrApp/Db/fru_basic_alias.template:
```diff
@@ -0,0 +1,27 @@
+include "fru_basic.db"
```
Upstream [`fru_common.db`][1]?

[1]: https://github.com/slac-epics-modules/ipmiComm/blob/R4.6.0/Db/fru_common.db

henriquesimoes reviewed (changes requested):
Theses substitutions turn out to be very confusing... I think I found some issues on the current implementation. Maybe I'm simply confusing things. If the latter turns out to be the case, it may payoff to further explain the logic in the commit message.

henriquesimoes commented on ipmiMgrApp/Db/microtca_bpm_crate.substitutions:
```diff
@@ -12,17 +12,12 @@
+            { "", "" }
 }
 
-#file "shelf.db"
-#{
-#    pattern { id }
-#            { Shelf }
-#}
-
```
This should be in another commit.

henriquesimoes commented on ipmiMgrApp/Db/shelf.substitutions:
If we don't load this file, why do we care about it?

Remove it along with its commented code in a standalone commit before this one.

henriquesimoes reviewed (approved):
Looks good for me. I haven't double-checked aliases names, but they should be fine if you replaced the `record(` by an `alias(` for specifying them all before adding the upstream-used names.

ericonr commented on ipmiMgrApp/Db/cooling_unit.substitutions:
```diff
@@ -23,19 +23,19 @@
 
-file "fru_cu_common.db"
+file "fru_common_alias.db"
 {
     pattern { id    , fruid    }
             { $(id) , $(fruid) }
 }
 
-file "fru_common.db"
+file "fru_cu_common_alias.db"
```
This diff is confusing because it's changing cu_common to common and vice-versa. Please reorder this block.

ericonr reviewed (commented):
Were `system_powerstate_unsupported`, `system_chassis_status`, `fru_extended` templates removed because they were included but not used? Or were they removed as part of the last commit? If the former, I think they should be cleaned up in a separate commit.

gustavosr8 committed:
Remove unused Db's

gustavosr8 committed:
Remove unused `shelf.db` record

gustavosr8 committed:
Change the locally defined Db's to use aliases from the ipmiComm db's

Since the defined Db's are just copies of ipmiComm db's, with the name
changed to fit the LNLS name convention, we can replace the record
definition with aliases to the support module record. Also add the macro
`$(sensinst)` in `sensor_ai_alias` that fixes a bug in the previous
implementation where records with the same name would be generated and
one would overwrite the other.
